### PR TITLE
Updates to AEX-2 Draft

### PR DESCRIPTION
**Pull request by shekhar-shubhendu**

### CODE REVIEW

File: AEXS/aex-2.md [hunk 1]
```diff
@@ -42,41 +42,43 @@ The specification is divided in two main areas, the first one addresses applicat
...
+#### Communication Examples
+
+##### Provider Registration (Handshake)
+![Register Provider][register]
+
+##### Transaction Signing (example operation)
+![Sign TX][sign]
+
+[sign]: https://i.imgur.com/C5cu3FI.png "Sign Tx"
+[register]: https://i.imgur.com/4qOLPDf.png "Register Provider"
```
Comment (knarz) [inline AEXS/aex-2.md]: Images can be included in the assets subdirectory

File: AEXS/aex-2.md [hunk 2]
```diff
@@ -85,7 +87,9 @@ notifiy the SDK (and the app) that the wallet has been disconnected.
 
 The wallet can also let the SDK know of extra funtionality that it provides during the time of registration. These methods will vary from wallet to wallet and are out of the scope of this proposal.
 
-#### Workflow
+#### Workflow (Messages Exchange Example)
+
+We're using [JSON-RPC 2.0](https://www.jsonrpc.org/specification#examples) standard for messages exchange (already supported by aeternity's JS SDK), but for simplicity's sake, this documentation is not using the standard, so the actual messages exchange is slightly different.
```
Comment (knarz) [inline AEXS/aex-2.md]: s/messages/message/

Comment (knarz) [reply AEXS/aex-2.md]: s/JSON-RPC 2.0 standard/the JSON-RPC 2.0 standard/

Comment (knarz) [inline AEXS/aex-2.md]: If you use the JSON-RPC standard then please also use it in this document. These documents should enable other implementers to produce an implementation without having to read much, if any, of the reference code

File: AEXS/aex-2.md [hunk 3]
```diff
@@ -101,32 +105,34 @@ The wallet can also let the SDK know of extra funtionality that it provides duri
...
      "registrationComplete": {
-       "provider": "<DH public key>",
-       "sdk": "<DH public key>"
+        "provider": "<DH public key>",
+        "sdk": "<DH public key>"
      }
    }
    ```
+
 3. At this step, the user should be prompted by the extension/wallet to accept the incoming SDK. If the user agrees, the wallet posts the `walletDetail` message encrypted using the secret key generated (wallet private key + sdk public key) and nonce.
```
Comment (knarz) [inline AEXS/aex-2.md:123]: It would help readability if you give the keys generated labels

File: AEXS/aex-2.md [hunk 4]
```diff
@@ -183,13 +189,13 @@ When the URI is invoked, if there is a default application selected by user for
   - `aeternity://<address>/<raw_tx>?network_id=<network_id>&callback=<sdk_url>`
 - Callback to SDK
 
-  - Every `sign` request should specify `sdk callback` url. Once the signing is done by the wallet, it needs to invode the callback url with the signex transaction payload for the `SDK` to broadcast it.
+  - Every `sign` request should specify `sdk callback` url. Once the signing is done by the wallet, it needs to invoke the callback url with the signex transaction payload for the `SDK` to broadcast it.
```
Comment (knarz) [inline AEXS/aex-2.md]: s/`sdk callback`/a `sdk callback`/

**Review by knarz — CHANGES_REQUESTED** [review]

File: AEXS/aex-2.md (hunk 4 above)
Comment (shekhar-shubhendu) [reply AEXS/aex-2.md]: fixed.

File: AEXS/aex-2.md (hunk 3 above)
Comment (shekhar-shubhendu) [reply AEXS/aex-2.md:123]: need more explanation or example.

File: AEXS/aex-2.md (hunk 2 above)
Comment (shekhar-shubhendu) [reply AEXS/aex-2.md]: updated.

Comment (shekhar-shubhendu) [reply AEXS/aex-2.md]: fixed.

File: AEXS/aex-2.md (hunk 1 above)
Comment (shekhar-shubhendu) [reply AEXS/aex-2.md]: updated.

**Review by knarz — APPROVED** [review]

**Pull request merged**